cxotime/cxotime.py

```diff
@@ -498,3 +501,47 @@ def to_value(self, parent=None, **kwargs):
         return out
 
     value = property(to_value)
+
+
+class CxoTimeDescr:
+    """Descriptor for an attribute that is CxoTime (in date format) or None if not set.
+
+    This allows setting the attribute with any CxoTimeLike value. The format of the
+    CxoTime value is set to "date" for uniformity of the representation.
+
+    If the attribute is set with a ``CxoTime`` object then a copy of the original is
+    made.
+
+    Note that setting this descriptor to ``None`` will set the attribute to ``None``,
+    which is different than ``CxoTime(None)`` which returns the current time. The
+    latter behavior is a compatibility feature with ``DateTime``.
+
+    Examples
+    --------
+    >>> from dataclasses import dataclass
+    >>> from cxotime import CxoTime, CxoTimeDescr
+    >>> @dataclass
+    ... class MyClass:
+    ...     time: CxoTime | None = CxoTimeDescr()
+    ...
+    >>> obj = MyClass(100)
+    >>> obj.time
+    <CxoTime object: scale='utc' format='date' value=1998:001:00:00:36.816>
+    >>> obj = MyClass()
+    >>> obj.time is None
+    True
+    """
+
+    def __set_name__(self, owner, name):
+        self._name = "_" + name
+
+    def __get__(self, obj, type):
+        if obj is None:
```

Comment:
When can obj be None?

Reply:
If you call this as a class attribute, e.g. `MyClass.time`.